x/stakeibc/keeper/delegation.go

```diff
@@ -10,26 +10,12 @@ import (
 	"github.com/cosmos/gogoproto/proto"
 	"github.com/spf13/cast"
 
-	icatypes "github.com/cosmos/ibc-go/v7/modules/apps/27-interchain-accounts/types"
-
 	"github.com/Stride-Labs/stride/v21/utils"
 	recordstypes "github.com/Stride-Labs/stride/v21/x/records/types"
 	"github.com/Stride-Labs/stride/v21/x/stakeibc/types"
 )
 
 func (k Keeper) DelegateOnHost(ctx sdk.Context, hostZone types.HostZone, amt sdk.Coin, depositRecord recordstypes.DepositRecord) error {
-	// TODO: Remove this block and use connection-id from host zone
```

Comment:
This change feels very reasonable, but do you know why we did it this way at all? I think `connection-id`'s always been on the host_zone struct?

Reply:
I think it's just a bit of tech debt! this was probably some of the first code written haha

My guess at what happens is:
*  We first wrote the `GetConnectionIdFromICAPort` function specifically for OnChanOpenAck 
    * maybe because weren't familiar with the IBC functions / didn't know you could map from channel ID to connection ID?
    *  I _think_ I used to believe that channel IDs were **not** globally unique and that they were only unique to each connection. Possible others thought this as well
* Then when were were writing functions like `DelegateOnHost`, `GetConnectionIdFromICAPort` seemed like the easiest helper to get what we wanted

Just speculating! But I've had it on my radar to fix this for awhile (hence the TODOs) and this was the first time I realized the lookup was actually incorrect!